test/tools/jsontests/StateTests.cpp

```diff
@@ -115,25 +116,49 @@ class generaltestfixture
 
 	void fillAllFilesInFolder(string _folder)
 	{
-		std::string fillersPath = dev::test::getTestPath() + "/src/GeneralStateTestsFiller/" + _folder;
+		std::string fillersPath = getTestPath() + "/src/GeneralStateTestsFiller/" + _folder;
+		using dirIt = boost::filesystem::directory_iterator;
 
-		boost::filesystem::directory_iterator iterator_tmp(fillersPath);
+		dirIt iterator_tmp(fillersPath);
 		int fileCount = 0;
-		for(; iterator_tmp != boost::filesystem::directory_iterator(); ++iterator_tmp)
+		for(; iterator_tmp != dirIt(); ++iterator_tmp)
 			if (boost::filesystem::is_regular_file(iterator_tmp->path()) && iterator_tmp->path().extension() == ".json")
 				fileCount++;
-		if (dev::test::Options::get().filltests)
+		if (Options::get().filltests)
 			fileCount *= 2; //tests are checked when filled and after they been filled
-		dev::test::TestOutputHelper::initTest(fileCount);
+		TestOutputHelper::initTest(fileCount);
 
-		boost::filesystem::directory_iterator iterator(fillersPath);
-		for(; iterator != boost::filesystem::directory_iterator(); ++iterator)
+		dirIt iterator(fillersPath);
+		for(; iterator != dirIt(); ++iterator)
 			if (boost::filesystem::is_regular_file(iterator->path()) && iterator->path().extension() == ".json")
 			{
 				string fileboost = iterator->path().filename().string();
-				dev::test::executeTests(fileboost, "/GeneralStateTests/"+_folder, "/GeneralStateTestsFiller/"+_folder, dev::test::doStateTests);
+				if (fileboost.rfind("BCFiller.json") != std::string::npos)
```

Comment:
so not to convert 100 of tests into blockchain we decided to add a marker to such tests to be run as blockchain but being state tests. 
I had an option in mind - to read file contents and then look for an option to run this test as blockchain.
then I realized that to do that I will have to parse whole json file and that would add to execution time. 
there for we have this hardcoded thing. 

the mere fact that state tests are being run as blockchain is a workaround. it was not designed to be run as blockchain tests.